eslint-local-rules.js

```diff
@@ -33,4 +33,42 @@ module.exports = {
       }
     },
   },
+  'no-async-module-init': {
+    meta: {
+      type: 'problem',
+      docs: {
+        description: 'disallow async module initialisation',
+        category: 'Possible Errors',
+        recommended: true,
+        url: '',
+      },
+      messages: {
+        noAsyncRegister:
+          'Disallowing async register functions in modules to prevent unexpected startup failures or timeouts.',
+        noAsyncProviderFactory:
+          'Disallowing async useFactory in module providers to prevent unexpected startup failures or timeouts.',
+      },
+    },
+    schema: [],
+    create: function (context) {
+      return {
+        "MethodDefinition[static=true][key.name='register'][value.async=true]": (
+          node,
+        ) => {
+          context.report({
+            node,
+            messageId: 'noAsyncRegister',
+          })
+        },
+        "ObjectExpression[properties.length=2] > Property[key.name='provide'] + Property[key.name='useFactory'][value.async=true]": (
```

Comment:
Think this selector is too narrow. It will not catch async factories where the provide property comes after, or has more properties like `scope` and `inject`.

It's probably ok to ban all async "useFactory" properties.

Reply:
Ahh of course, fixed.